compiler/compile/ResolvedMethod.cpp

```diff
@@ -19,3 +19,636 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
+#include "compile/ResolvedMethod.hpp"
+
+#include <string.h>
+#include "compile/Compilation.hpp"
+#include "compile/Method.hpp"
+#include "compile/SymbolReferenceTable.hpp"
+#include "control/Options.hpp"
+#include "control/Options_inlines.hpp"
+#include "env/jittypes.h"
+#include "il/SymbolReference.hpp"
+#include "il/symbol/ParameterSymbol.hpp"
+#include "il/symbol/ResolvedMethodSymbol.hpp"
+#include "infra/List.hpp"
+#include "runtime/Runtime.hpp"
+
+class TR_FrontEnd;
+class TR_OpaqueMethodBlock;
+class TR_PrexArgInfo;
+namespace TR { class IlGeneratorMethodDetails; }
+namespace TR { class LabelSymbol; }
+
+bool TR_ResolvedMethod::isDAAMarshallingWrapperMethod()
+   {
+#ifdef J9_PROJECT_SPECIFIC
+   if (getRecognizedMethod() == TR::com_ibm_dataaccess_ByteArrayMarshaller_writeShort        ||
+       getRecognizedMethod() == TR::com_ibm_dataaccess_ByteArrayMarshaller_writeShortLength  ||
```

Comment:
Does this mean we can make a change over in OpenJ9 to override this method and eliminate this `J9_PROJECT_SPECIFIC` as well as similar others in this file?

Reply:
Sorry, I missed this comment from earlier.

One of the eventual goals is to make the base of the ResolvedMethod hierarchy an extensible class so that project-specific functions such as you pointed out can be encapsulated in the project they should belong in.  Our starting point was the base `Method` class, and this PR starts to put the framework in place for downstream projects to start sinking their project-specific specializations into the concrete `TR::Method` class that will be used throughout OMR and downstream projects.  A further goal is to provide some conceptual integrity documentation for what it means to be a Method.

Once the base Method hierarchy has been dealt with we'll turn our attention to addressing the ResolvedMethod hierarchy.  There is a definite base to this hierarchy that can be made into an extensible class with project-specific code (and this is where the methods you highlighted above should likely live in the OpenJ9 project), but derived from that base is another "resolved" hierarchy that draws from both Method and ResolvedMethod.  Some projects based on OMR use a `TR::ResolvedMethod` class to refer to this additional "resolved" hierarchy that sits at the bottom of the hierarchy.  OpenJ9 takes a different approach and assumes the base of the hierarchy `TR_ResolvedMethod` is passed around and that methods can be invoked from that. You may recall that internally within IBM a couple of years back we had some discussions on the semantics of this hierarchy and there were a number of opinions.   We will need to be careful to sort this out to add purpose and meaning to each class.  I will restart this discussion in the community when the time comes, seeding it with my notes of that discussion we had internally as a starting point.

Reply:
Thanks for the in-depth explanation. I'll leave this unresolved for future reference so it remains visible to others.